fundamentals/fundamentals-bio/src/main/scala/izumi/functional/bio/data/FreeInterpreter.scala

```diff
@@ -0,0 +1,20 @@
+package izumi.functional.bio.data
+
+/**
+  * Should executed scenario steps in a single context and save all scenario metrics.
+  */
+trait FreeInterpreter[F[+_, +_], S[_[+_, +_], _, _], Scope] {
+  /** Single step execution. */
+  def interpret[E, A](scope: Scope, s: S[F, E, A]): F[Nothing, FreeInterpreter.Result[F, S, Scope]]
```

Comment:
We seem to be missing an equivalent to `foldMap`[[1]](https://github.com/typelevel/cats/blob/master/free/src/main/scala/cats/free/Free.scala#L158)[[2]](https://typelevel.org/cats/datatypes/freemonad.html#run-your-program)
That is a function like
```scala
def foldMap[F[+_, +_], G[+_, +_], E, A](f: Free[F, E, A])(g: F ~> G)(implicit G: BIOPanic[G]): G[E, A]
```
That would convert inner `F` in Free to G and convert the branches of Free to G, thereby converting everything to G